rabit/CMakeLists.txt

```diff
@@ -4,7 +4,7 @@ find_package(Threads REQUIRED)
 
 set(RABIT_SOURCES
   ${CMAKE_CURRENT_LIST_DIR}/src/allreduce_base.cc
-  ${CMAKE_CURRENT_LIST_DIR}/src/c_api.cc)
+  ${CMAKE_CURRENT_LIST_DIR}/src/rabit_c_api.cc)
```

Comment:
msvc complains 2 files having same name might produce incorrect result.  So I renamed the c api file and merged  the test_io.cc into xgboost.